pytest_fixtures/component/provision_azure.py

```diff
@@ -9,6 +9,45 @@
 from robottelo.constants import AZURERM_RHEL7_FT_GALLERY_IMG_URN
 from robottelo.constants import AZURERM_RHEL7_FT_IMG_URN
 from robottelo.constants import AZURERM_RHEL7_UD_IMG_URN
+from robottelo.constants import DEFAULT_ARCHITECTURE
+
+
+@pytest.fixture(scope='session')
+def sat_azure(request, session_puppet_enabled_sat, session_target_sat):
+    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
+    yield hosts[request.param]
+
+
+@pytest.fixture(scope='module')
+def sat_azure_org(sat_azure):
+    yield sat_azure.api.Organization().create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_loc(sat_azure):
+    yield sat_azure.api.Location().create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_domain(sat_azure, sat_azure_loc, sat_azure_org):
+    yield sat_azure.api.Domain(location=[sat_azure_loc], organization=[sat_azure_org]).create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_default_os(sat_azure):
+    """Default OS on the Satellite"""
+    search_string = 'name="RedHat" AND (major="6" OR major="7" OR major="8" OR major="9")'
```

Comment:
Not a subject of this PR, I just want to bring up some attention to hardcoding RHEL versions in our tests. Once the new RHEL version arrives, we should ideally bump all of our test cases to include the new RHEL version. I wish we have a dynamic way to do so. We already have `settings.supportability.content_hosts.rhel.versions` which we could make use of. I would like to see something like:
```python
supported_rhel_vers = [ver for ver in settings.supportability.content_hosts.rhel.versions if isinstance(ver, int)]
search_string = ''.join([ f'major="{v}"{"" if v == supported_rhel_vers[-1] else " OR "}' for v in supported_rhel_vers])
```

What are your thoughts about this, @jyejare ? Can you think of any helpers that test owners could use for this purpose?

Reply:
Or this one:
```python
# the following could be some kind of robottelo helper function
supported_rhel_vers = [ver for ver in settings.supportability.content_hosts.rhel.versions if isinstance(ver, int)]

' OR '.join(map(lambda x: f'major="{x}"', supported_rhel_vers))
```

Reply:
@jyejare gentle ping, just not to burry this conversation thread entirely as you merged the PR already.

Reply:
Replied in GChat and we have created a new issue for it Robottelo to discuss about it #10481